docs/source/java/flight_sql_jdbc_driver.rst

```diff
@@ -115,18 +115,20 @@ parameters are:
      - When TLS is enabled, the password for the certificate store
 
    * - useEncryption
-     - false
-     - Whether to use TLS (the default is an insecure, plaintext
-       connection)
+     - true
```

Comment:
Maybe discuss if this is case sensitive? When I was setting up DBeaver, all the other major parameters were in the form FOO_BAR, and this one is documented as fooBar

Reply:
Hmm, all parameters should be camelCase per:
https://github.com/apache/arrow/blob/547b2406d71bf9ad14e2bcc7064c6923165dd737/java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/utils/ArrowFlightConnectionConfigImpl.java#L168
and
https://github.com/apache/arrow/blob/547b2406d71bf9ad14e2bcc7064c6923165dd737/java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/utils/ArrowFlightConnectionConfigImpl.java#L202

Though the get() implementation seems to also allow fully lower-case. It's probably supposed to be case-insensitive though.

Reply:
Let's explicitly document whether it's case-sensitive or not, and maybe we can split out a separate ticket to make it case-insensitive if needed

Reply:
I've documented that names are case-sensitive for simplicity's sake.

If we want to be more precise, names are case-insensitive on the connection URI (since we internally lower-case them after parsing) and case-sensitive in the Properties object, but really they should consistently be case-insensitive.